std/haxe/http/HttpBase.hx

```diff
@@ -185,4 +221,30 @@ class HttpBase {
 		`httpInstance.onStatus = function(status) { // handle status }`
 	**/
 	public dynamic function onStatus(status:Int) {}
+
+	/**
+		Override this if extending `haxe.Http` with overriding `onData`
+	**/
+	function hasOnData():Bool {
+		return !Reflect.compareMethods(onData, emptyOnData);
```

Comment:
I wonder if we really need all this `emptyOnData` trickery, couldn't we "just" rework this to `var onData/onBytes/onStatus` and check for `null` here and there?

Reply:
That would break things if someone extends `Http` and overrides `onData` instead of assigning a callback to it.
But if @ncannasse is ok with it, I'll change to `var`s. I agree that's better.

Reply:
> overrides `onData` instead of assigning a callback to it.

O_o that works?

Reply:
http://try-haxe.mrcdk.com/#0e559